### PR TITLE
Chapters fade when all its challenges are completed

### LINKED ISSUE #6648: Map's completed chapterBlocks are not faded
When displaying the map, chapters are not being faded, unless all the siblings inside the same certification have their challenges completed.

Working on a fix, but just thought I'd document it here to avoid duplicates later.

screen shot 2016-02-01 at 09 25 36

------

Also, as noticed in the following image, the links remain underlined after click. fixing that too
screen shot 2016-02-01 at 09 21 23

------

Will comment with PR link when ready.

### PR DESCRIPTION
**Pull request by justincorrigible**
It wasn't doing it unless all the chapters in a same certification had all the others challenges completed too. fixed. :)

---

Also, removing the decoration after click.

closes #6648

### CODE REVIEW

File: client/main.js [hunk 1]
```diff
@@ -309,10 +309,12 @@ $(document).ready(function() {
   });
 
   $.each($('.map-collapse'), function(i, div) {
+    console.log(div, $(div).find('.manip-hidden').length,
+        $(div).find('p').length);
```
Comment (bugron) [inline client/main.js]: Please, remove this `console.log`.

**Comment by bugron** [comment]
@BerkeleyTrue can you take a look at Travis's error https://travis-ci.org/FreeCodeCamp/FreeCodeCamp/builds/106269480#L2714?

**Comment by bugron** [comment]
@hallaathrad ?

**Comment by bugron** [comment]
This PR passes all tests in my local. Looks good.

**Comment by BerkeleyTrue** [comment]
@bugron Did you try with a fresh npm install?

File: client/less/map.less [hunk 2]
```diff
@@ -91,16 +91,15 @@
...
-.map-buttons .input-group{
-  width:300px;
-}
-
-.map-buttons .input-group{
-  margin-top: 15px;
-  margin-left: auto;
-  margin-right: auto;
+  & button,
+  & .input-group{
```
Comment (SaintPeter) [inline client/less/map.less]: `&`?

Comment (hallaathrad) [inline client/less/map.less]: it's a preprocessor function to summon the parent... 

You can also bundle pseudo-selectors with your mixins using this method. Here's the classic clearfix hack, rewritten as a mixin (& represents the current selector parent):

```CSS
.clearfix {
  display: block;
  zoom: 1;

  &:after {
    content: " ";
    display: block;
    font-size: 0;
    height: 0;
    clear: both;
    visibility: hidden;
  }
}
```

**Comment by bugron** [comment]
@BerkeleyTrue no, should I?

**Comment by BerkeleyTrue** [comment]
Yes, in order to find the cause of the break in travis.

**Comment by SaintPeter** [comment]
:+1:

**Pull request merged**